services/meta/data.go

```diff
@@ -465,8 +466,33 @@ func (data *Data) DropContinuousQuery(database, name string) error {
 	return ErrContinuousQueryNotFound
 }
 
+// validateURL returns an error if the URL does not have a port or uses a scheme other than UDP or TCP.
+func validateURL(input string) error {
+	u, err := url.Parse(input)
+	if err != nil {
+		fmt.Println(err)
+		return ErrInvalidSubscriptionURL(input)
+	}
+
+	if u.Scheme != "udp" && u.Scheme != "tcp" {
+		return ErrInvalidSubscriptionURL(input)
+	}
+
+	hostParts := strings.Split(u.Host, ":")
```

Comment:
The issue addresses having an invalid scheme/protocol but using udp without knowing the port number is also not correct.

Currently it just requires a non-empty string to be set for the port. This is related to my questions about default port numbers and tcp vs http. Once I better understand the requirements this will most likely change, at a minimum it could verify the port is a positive integer.